### PR TITLE
Created API Skeleton/Template

### PR DESCRIPTION
**Pull request by johnsimeroth**
Changes in this PR:

- added axios to dependencies list
- added api file and folder structure
- created "template" functions for get, post, put requests that can be reused for (I hope) all of our individual functions.
-    ^open for discussion: we could just use these templates directly in our code, but I think they'll be less readable.
- created all reviews API functions.
- created the first "product" API function
- added env directory with an example config file that we will each need to fill out for API to work. Included instructions in the config file.

### CODE REVIEW

File: client/src/lib/api.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// This first section contains helper/template functions to keep our other functions
+// simple and DRY. Reuse them! But please don't edit them without team buy-in, because all
+// of our individual API functions rely on them.
+const api = {};
+
+const url = `https://app-hrsei-api.herokuapp.com/api/fec2/${config.CAMPUS_CODE}`;
+
+const get = (endpoint, params = {}) => {
+  return axios.get(`${url}/${endpoint}/`, {headers: {Authorization: config.AUTH}, params: params})
```
Comment (brawoods) [inline client/src/lib/api.js:13]: I think we should consider using NPM paths here to join the url and endpoints instead to work around any problems we may run into hardcoding separations like "/".

File: client/src/lib/api.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    .then(results => results.data)
+    .catch(err => console.error(`Error getting ${endpoint} from server:  ${err}`));
+}
+
+const post = (endpoint, data = {}, params = {}) => {
+  return axios.post(`${url}/${endpoint}/`, data, {headers: {Authorization: config.AUTH}, params: params})
+    .then(results => results.data)
+    .catch(err => console.error(`Error getting ${endpoint} from server:  ${err}`));
+}
+
```
Comment (brawoods) [inline client/src/lib/api.js:23]: Why are we returning results.data? What data will there be to return for a post?

File: client/src/lib/api.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const put = (endpoint, data = {}, params = {}) => {
+  return axios.put(`${url}/${endpoint}/`, data, {headers: {Authorization: config.AUTH}, params: params})
+    .then(results => results.data)
+    .catch(err => console.error(`Error getting ${endpoint} from server:  ${err}`));
+}
+
```
Comment (brawoods) [inline client/src/lib/api.js:29]: Same comment for PUT as POST.

**Review by brawoods — APPROVED** [review]
Left a few comments and questions, but didn't see anything that should prevent a merge.

**Comment by brawoods** [comment]
Something I missed, but want to note so we don't forget. If you have the tests you ran for the code, try and save it or note it somewhere so we can add it to our test library once it's up and running.

File: client/src/lib/api.js (hunk 1 above)
Comment (johnsimeroth) [reply client/src/lib/api.js:13]: good idea! I'm open to that. I can work on that in v2.

File: client/src/lib/api.js (hunk 2 above)
Comment (johnsimeroth) [reply client/src/lib/api.js:23]: Sometimes post requests will return the db version of the thing you just posted.. e.g. if you add a new review and then when you send it to the database, it automatically generates a unique id and timestamp, I like to immediately "get" that back from the database to keep my client object and my db object formats in sync.

File: client/src/lib/api.js (hunk 3 above)
Comment (johnsimeroth) [reply client/src/lib/api.js:29]: same reasoning as above.

File: client/src/lib/api.js (hunk 2 above)
Comment (brawoods) [reply client/src/lib/api.js:23]: Okay, got it. That makes sense. Good thinking!

File: client/src/lib/api.js (hunk 3 above)
Comment (brawoods) [reply client/src/lib/api.js:29]: same as above.

**Pull request merged**